### DISCUSSION TITLE
Netbox - Datasource - Error - UnboundLocalError("local variable '_' referenced before assignment")

### DISCUSSION THREAD

**Discussion by alainseys** [netbox-community/netbox#15893] [Q&A] [answered]
Hello, 

i want to add a git repo as a datasource for the scripts , i can add this but there is no option to add a ssh key what i find weird.
Even without that if i start sync it failed to sync and throws me this error.

`UnboundLocalError("local variable '_' referenced before assignment")`

i have tried with ssh and https both do not work.
Even tried with https and PAT nu luck.

How can i debug this further ? 

I have looked in to the documentation pages but nothing much there :) 
[https://ipam.vanmarcke.be/static/docs/models/core/datasource/](url)

**Comment by markkuleinio** [comment]
Workaround to use SSH key: https://github.com/netbox-community/netbox/issues/15394#issuecomment-2016409907

**Reply by alainseys** [reply]
/etc/systemd/system/netbox-rq.service, is not present on my system

**Comment by markkuleinio** [comment]
Have you installed `dulwich`?

https://github.com/netbox-community/netbox/issues/15510

https://docs.netbox.dev/en/stable/features/synchronized-data/:

> Data backends which connect to external sources typically require the installation of one or more supporting Python libraries. The Git backend requires the [dulwich](https://www.dulwich.io/) package, and the S3 backend requires the [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html) package. These must be installed within NetBox's environment to enable these backends.

**Reply by alainseys** [reply]
i dont think so i have followed these step to get started with netbox 

[https://github.com/netbox-community/netbox/tree/develop/docs/installation](url)

but i will try the ssh key

**Reply by alainseys** [reply]
Just checked i have installed dulwich on my system trough the python venv

**Reply by alainseys** [reply, marked as answer]
ok i think i have forgotten to upgrade after upgrading this now works 

` sudo /opt/netbox/upgrade.sh
`

**Reply by alainseys** [reply]
@markkuleinio thanks for the help
